Intro (#51)
infinum/eightshift-docs: website/docs/basics/extending-classes.md

---
id: extending-classes
title: Extending Classes
sidebar_label: Extending Classes
---

[![docs-source](https://img.shields.io/badge/source-eigthshift--libs-blue?style=for-the-badge&logo=php&labelColor=2a2a2a)](https://github.com/infinum/eightshift-libs)

> Every class in the lib can be extended. To follow the good practice you should never use class directly from the Eightshift-libs in you service container, but rather you should create an class in your project and extend the class from the Eightshift-libs. That is why we created all those [WP-CLI](wp-cli) commands that you saw in the previous chapter.

Reviewer: ```suggestion
Every class in the lib can be extended. To follow the good practice, you should never use the class directly from the Eightshift-libs in your service container, but rather you should create a class in your project and extend the class from the Eightshift-libs. That is why we created all those [WP-CLI](wp-cli) commands that you saw in the previous chapter.
```

> We don't do any magic in the Eightshift-libs like in the previous versions. All classes contains only methods that you need to call using WordPress hooks or filters.

Reviewer 2: ```suggestion
We don't do any magic in the Eightshift-libs, like we did in previous versions. All classes contain only methods that you need to call using WordPress hooks or filters.
```

> We don't do any magic in the Eightshift-libs like in the previous versions. All classes contains only methods that you need to call using WordPress hooks or filters.

Reviewer: ```suggestion
We don't do any magic in the Eightshift-libs like in the previous versions. All classes contain only methods that you need to call using WordPress hooks or filters.
```

## Example

> Let's look at a way to add Class that enqueues theme frontend scripts and styles.

Reviewer 2: ```suggestion
Let's look at a way to add class that enqueues themes frontend scripts and styles.
```

> Let's look at a way to add Class that enqueues theme frontend scripts and styles.

Reviewer: ```suggestion
Let's look at a way to add class that enqueues theme frontend scripts and styles.
```

> Go to the **root** of your theme using the terminal and run this command:

Reviewer: ```suggestion
Go to the **root** of your theme, and using the terminal, run this command:
```

`wp boilerplate create_enqueue_admin`

> This command will output something like this in your project, taking care of your project namespace and vendor prefixes, as well as package name and text-domain:

Reviewer: ```suggestion
This command will output something like this in your project, taking care of your projects namespace and vendor prefixes, as well as the package name and the text-domain:
```

> ```js

Reviewer 2: ```suggestion
```php
```

Author: It doesn't style good.

use CoolProjectVendor\EightshiftLibs\Manifest\ManifestInterface;

/**
 * Class EnqueueAdmin
 *
 * This class handles enqueue scripts and styles.
 */
class EnqueueAdmin extends AbstractEnqueueAdmin
{

	/**
	 * Create a new admin instance.
	 *
	 * @param ManifestInterface $manifest Inject manifest which holds data about assets from manifest.json.
	 */
	public function __construct(ManifestInterface $manifest)
	{
		$this->manifest = $manifest;
	}

	/**
	 * Register all the hooks
	 *
	 * @return void
	 */
	public function register(): void
	{
		add_action('login_enqueue_scripts', [ $this, 'enqueueStyles' ]);
		add_action('admin_enqueue_scripts', [ $this, 'enqueueStyles' ], 50);
		add_action('admin_enqueue_scripts', [ $this, 'enqueueScripts' ]);
	}

	/**
	 * Method that returns assets name used to prefix asset handlers.
	 *
	 * @return string
	 */
	public function getAssetsPrefix(): string
	{
		return Config::getProjectName();
	}

	/**
	 * Method that returns assets version for versioning asset handlers.
	 *
	 * @return string
	 */
	public function getAssetsVersion(): string
	{
		return Config::getProjectVersion();
	}
}
```

> As you see from the provided example we created a new class and extended the functionality from the Eightshift-libs.

Reviewer: ```suggestion
As you can see from the provided example, we created a new class and extended the functionality from the Eightshift-libs.
```

> In your new class you have `register method` that must be in every class that registers WordPress action hooks or filters, and we call them **service classes**.

Reviewer 2: ```suggestion
In your new class you have a `register` method that must be in every class that registers WordPress action hooks or filters, and we call them **service classes**.
```

> In your new class you have `register method` that must be in every class that registers WordPress action hooks or filters, and we call them **service classes**.

Reviewer: ```suggestion
In your new class, you have a `register` method that must be in every class that registers WordPress action hooks or filters. We call these **service classes**.
```

> Your class contains three action hooks now, from this example you can see that if you don't want to use some functionality just remove the action and it is not used any more. For more details what each hooks callback does just go to the extended class in Eightshift-libs and see what it does.

Reviewer: ```suggestion
Your class contains three action hooks now. From this example, you can see that if you don't want to use some functionality just remove the action and it is not used anymore. For more details on what each hook's callback does, just go to the extended class in Eightshift-libs and see the logic defined there.
```

You can extend every class from the library. To follow the good practice, you should never use the class directly from the Eightshift-libs in your service container; but instead, you should create a class in your project and extend the class from the Eightshift-libs. That is why we made all those [WP-CLI](wp-cli) commands that you saw in the previous chapter.

> We don't do any magic in the Eightshift-libs as we did in previous versions. All classes contains only methods that you need to call using WordPress hooks or filters.

Reviewer: ```suggestion
We don't do any magic in the Eightshift-libs as we did in previous versions. All classes contain only the methods that you need to call using WordPress hooks or filters.
```

## Example

> Let's look at a way to add a class that enqueues themes frontend scripts and styles.

Reviewer: ```suggestion
Let's look at a way to add a class that enqueues theme frontend scripts and styles.
```
